### ISSUE TITLE
UE4Editor-HaxeInit.dll missing or built with different engine version

### ISSUE THREAD

**Issue by Chlorodatafile** [proletariatgames/unreal.hx#30]
Hello !

I am actually trying to setup my first Haxe project with Unreal Engine, since one week, but I still encounter the same issue : "UE4Editor-HaxeInit.dll missing or built with different engine version".

It freeze totally the execution by the impossibility to finish the compilation.

I've try on version 4.10.1 first, but I've seen that the plugin was made for 4.8.0, so I've installed the 4.8.3, because it was the only 4.8 version available for the download.

So, how can I solve that ?

**Comment by waneck** [comment]
Did you run `haxe init-plugin.hxml`?

**Comment by Chlorodatafile** [comment]
Yes, I did, and I have no error during the execution.
```
   InitPlugin.hx:22: Inferring project location...
   InitPlugin.hx:13: Game location: ../../Paralights_UnrealHx.uproject
   InitPlugin.hx:48: Updating game project...
   InitPlugin.hx:50: Updating game module...
   InitPlugin.hx:140: Building BuildTool...
   InitPlugin.hx:18: Project update done.
```

**Comment by waneck** [comment]
Try to build the first time on the command-line:
```[path-to-unreal]/Engine/Build/BatchFiles/[Your-Platform]/Build[.sh|.bat] [path-to-your-project]/Paralights_UnrealHx.uproject Paralights_UnrealHxEditor [Your-Platform] Development```

For example for Linux it would be something like:
```[path-to-unreal]/Engine/Build/BatchFiles/Linux/Build.sh [path-to-your-project]/Paralights_UnrealHx.uproject Paralights_UnrealHxEditor Linux Development```
Windows' paths are a little different (I think it's directly on `BatchFiles`, and the platform name is `Win64`)

**Comment by Chlorodatafile** [comment]
I've just try, with Win64 in BatchFiles, and just had that : http://pastebin.com/RSaiyH0W

I may should try on linux, I can install one on the computer who run Unreal to try if needed.

**Comment by waneck** [comment]
I think this is a problem of using the latest VSC with hxcpp. Can you try an earlier Visual Studio version? Or try updating your `hxcpp`

**Comment by waneck** [comment]
Never mind about updating `hxcpp`. But I think that running it with an earlier VSC++ should work. We are testing `unreal.hx` on Visual Studio 2013. Which version are you using?

**Comment by Chlorodatafile** [comment]
I've installed the 2013 version for the 4.8.3 engine version, so I'm actually searching why it's not the good version who run the compilation.

**Comment by waneck** [comment]
I think it's hxcpp which finds this version. Also note that we're developing on UE 4.9 . 4.8.x should work as well, but we aren't testing it atm.
We will look into how to make it work with latest VS, but for now you can uninstall the latest VS to make this work

**Comment by Chlorodatafile** [comment]
You work on 4.9 ? I will install it so, I've searched which version you use but it's not written anywhere except on the UE4Haxe.plugin.

If it's a conflict issue, I'll uninstall 4.10, and VS15. I will keep you in touch of how it evolve. :)

**Comment by waneck** [comment]
Okay! Keep us posted!

**Comment by Chlorodatafile** [comment]
Think to write somewhere that actually it's make for the 4.9 version of the engine and VS13, that will help the other developer who want to try to know which version to use. ;)

**Comment by waneck** [comment]
Yeah, we'll do it before launch - also I plan to simplify the setup steps so it's as smooth as possible

**Comment by Chlorodatafile** [comment]
If I can be of any kind of help, I would happily help you with that. It's a very excellent project and it will help to promote the video game development. :)

**Comment by waneck** [comment]
Trying it out and reporting back any issues is already a great help :)

**Comment by Chlorodatafile** [comment]
So, I've retry, the init-plugin say the same things, and this time I get that during the execution of the second command : 
```
Building UnrealHeaderTool...
Creating makefile for UnrealHeaderTool (no existing makefile)
ERROR: Couldn't find target rules file for target 'UnrealHeaderTool' in rules assembly 'UE4ModuleRules, Version=0.0.0.0,
 Culture=neutral, PublicKeyToken=null'.
Location: G:\Dev\UE4\Epic Games\4.9\Engine\Intermediate\Build\BuildRules\UE4ModuleRules.dll
Target rules found:
        UE4Editor - G:\Dev\UE4\Epic Games\4.9\Engine\Source\UE4Editor.Target.cs
        UE4Game - G:\Dev\UE4\Epic Games\4.9\Engine\Source\UE4Game.Target.cs
UnrealHeaderTool failed for target 'Paralights_UnrealHx' (platform: Win64, module info: C:\Development\Unreal\Paralights
_UnrealHx\Intermediate\Build\Win64\Paralights_UnrealHx\Development\UnrealHeaderTool.manifest).
HaxeModuleRules.hx:72: Touching C:\Development\Unreal\Paralights_UnrealHx\Source\Paralights_UnrealHx\Paralights_UnrealHx
.Build.cs
```

**Comment by waneck** [comment]
You're using a downloaded editor, right? If so, add -rocket to the command
line args

**Comment by Chlorodatafile** [comment]
I've used the command with -rocket, no error, so I've launch the project and still had the problem, but, this time, when I've rebuild the UE4Editor-HaxeInit.dll, it work without any kind of error ! So it's solved for the moment, I've many things to try and if I have another problem, it would not be linked directly to this problem. :)

Thanks you for your help ! :D

**Comment by waneck** [comment]
Thanks. But actually there is a problem with VS 2015 that I think I should look into.

**Comment by tanis2000** [comment]
I tried running the same on Mac but here's the result:

```
iMacValerio:4.10 tanis$ pwd
/Users/Shared/UnrealEngine/4.10
iMacValerio:4.10 tanis$ Engine/Build/BatchFiles/Mac/Build.sh /Users/tanis/Documents/Unreal\ Projects/UE4Hx-Tests/HaxeUnitTests.uproject HaxeUnitTestsEditor Mac Development
Engine/Build/BatchFiles/Mac/Build.sh: line 10: [: too many arguments
Building UBT...
MSBUILD: error MSBUILD0000: Project file 'Engine/Source/Programs/UnrealBuildTool/UnrealBuildTool.csproj' not found.
Building /Users/tanis/Documents/Unreal Projects/UE4Hx-Tests/HaxeUnitTests.uproject...
2016-01-11 15:26:09.287 defaults[12459:711497]
The domain/default pair of (/Users/tanis/Library/Preferences/com.apple.dt.Xcode, IDEBuildOperationMaxNumberOfConcurrentCompileTasks) does not exist
Running command : Engine/Binaries/DotNET/UnrealBuildTool.exe /Users/tanis/Documents/Unreal Projects/UE4Hx-Tests/HaxeUnitTests.uproject HaxeUnitTestsEditor Mac -deploy Development
Creating makefile for Unreal (no existing makefile)
WARNING: UnrealBuildTool found an Intermediate folder while looking for rules '/Users/Shared/UnrealEngine/4.10/Engine/Source/Projects/UE4Hx-Tests/Intermediate'.  It should only ever be searching under 'Source' folders -- an Intermediate folder is unexpected and will greatly decrease iteration times!
ERROR: Couldn't find target rules file for target 'Unreal' in rules assembly 'HaxeUnitTestsModuleRules, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
Location: /Users/Shared/UnrealEngine/4.10/Engine/Source/Projects/UE4Hx-Tests/Intermediate/Build/BuildRules/HaxeUnitTestsModuleRules.dll
Target rules found:
	UE4Editor - /Users/Shared/UnrealEngine/4.10/Engine/Source/UE4Editor.Target.cs
	UE4Game - /Users/Shared/UnrealEngine/4.10/Engine/Source/UE4Game.Target.cs
```

**Comment by waneck** [comment]
Fixed the issue with VS 2015 at 09d6379a5685a3d636e53f7ff0ab40f4fceaaf99 .

@tanis2000 , please check the ant build scripts at https://github.com/waneck/HaxePlatformerGame and try using them. They should work on Windows, Mac and Linux
